packages/langium/src/parser/async-parser.ts

```diff
@@ -30,7 +30,7 @@ export interface AsyncParser {
      *
      * @throws `OperationCancelled` if the parsing process is cancelled.
      */
-    parse<T extends AstNode>(text: string, cancelToken: CancellationToken): Promise<ParseResult<T>>;
+    parse<T extends AstNode>(text: string, options: ParserOptions | undefined, cancelToken: CancellationToken): Promise<ParseResult<T>>;
```

Comment:
The PR contains multiple "minor" breaking changes, but in sum I'd say the changes are significant enough to move this to v4.0. Shall we start planning the new major version more concretely?